### PR TITLE
Update Substrate to version 2.0.1

### PR DESCRIPTION
**Pull request by c410-f3r**
Unbreaks the building of:

* Newer nightly compilers
* New release of `futures` 0.3.9
* New release of `syn` 1.0.58

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2020-10-05
\ No newline at end of file
+nightly-2022-01-10
```
Comment (olanod) [inline rust-toolchain:1]: Do we need this? how does this work specifying a future `2022` version?

**Comment by olanod** [comment]
I'm very glad the nightly thing got fixed!

File: rust-toolchain (hunk 1 above)
Comment (c410-f3r) [reply rust-toolchain]: Nah, it was a mistake

**Comment by olanod** [comment]
Not so much of a "patch" release?

**Comment by c410-f3r** [comment]
Yeah. There are many things involved and the actual usage of newer compilers is blocked on https://github.com/paritytech/substrate-subxt/pull/217

**Pull request merged**